pyproject.toml

```diff
@@ -55,6 +54,7 @@ repository = "https://github.com/DiamondLightSource/HTTomo"
 [project.optional-dependencies]
 tomopy = [ "tomopy" ]
 httomolib = [ "httomolib" ]
+cupy = ["cupy"]
```

Comment:
Since only the httomolib backend requires cupy for our project at the moment, should we add cupy dependency to it?

```toml
httomolib = ["httomolib", "cupy"]
```